src/runtime/components/forms/Toggle.vue

```diff
@@ -97,10 +107,13 @@ export default defineComponent({
 
     const active = computed({
       get () {
-        return props.modelValue
+        if (typeof props?.onValue === 'object') {
+          return isEqual((props.modelValue as Record<string, unknown> | unknown[]), (props.onValue as Record<string, unknown> | unknown[]))
```

Comment:
You should use `isEqual` from `ohash`, it's already installed and used by the `Link` component.